SingleAxisStage.cpp

```diff
@@ -357,3 +550,59 @@ SingleAxisStage::MakeName(MotorDrive* motorDrive) const {
 
     return name;
 }
+
+int 
+SingleAxisStage::OnStageNameChange(MM::PropertyBase* pProp, MM::ActionType eAct)
+{
+    if (eAct == MM::BeforeGet)
+    {
+    }
+    else if (eAct == MM::AfterSet)
+    {
+        //Allows for the stage selection to update the displayed pre-init properties. 
+        //Can be useful for making adjustments to existing profiles when switching back to custom input
+
+        char buf[MM::MaxStrLength];
+        (void)GetProperty(PROP_StageNameSelection, buf);
+
+        selectedStageName_ = std::string(buf);
+
+        if (selectedStageName_.compare(PROPVAL_StageNameCustom) != 0 && selectedStageName_.compare(PROPVAL_StageNameAuto) != 0 && selectedStageName_.compare(PROPVAL_StageNameDEFAULT))
```

Comment:
Do you think it would be possible to load the setting values also when AUTO is selected? That would be more intuitive than the values staying the same but (presumably) being overridden by the detected actuator.

Reply:
Deferred because stage hw may take precedence over values in XML.